lm_eval/evaluator.py

```diff
@@ -112,7 +112,17 @@ def simple_evaluate(
             + "_rank" + str(lm.rank) + ".db",
         )
 
-    task_dict = lm_eval.tasks.get_task_dict(tasks, num_fewshot=num_fewshot)
+    task_dict = lm_eval.tasks.get_task_dict(tasks)
+    for task_name in task_dict.keys():
+        config = task_dict[task_name]._config
+        if num_fewshot is not None:
+            if config["num_fewshot"] > 0:
+                default_num_fewshot = config["num_fewshot"]
+                eval_logger.warning(
+                    f"Overwriting default num_fewshot of {task_name} from {default_num_fewshot} to {num_fewshot}"
+                )
```

Comment:
@haileyschoelkopf The first part of what you mentioned is implemented here. If the user doesn't set a `num_fewshot` value and it's not also in the yaml file, the default value will be 0 (this is from the Config class)